test/utils_test.rb

```diff
@@ -58,4 +58,36 @@ def test_forked_process_id_collision
     refute_equal(Datadog::Utils.next_id, r.read.chomp.to_i)
     r.close
   end
+
+  def test_utf8_encoding_happy_path
+    # we can't use utf-8 literals because our tests run against ruby 1.9.3
+    str = "pristine \U+FFE2".encode(Encoding::UTF_8)
+
+    assert_equal("pristine \U+FFE2", Datadog::Utils.utf8_encode(str))
+
+    assert_equal(::Encoding::UTF_8, Datadog::Utils.utf8_encode(str).encoding)
+
+    # we don't allocate new objects when a valid UTF-8 string is provided
```

Comment:
Great test! this ensures that we don't introduce any performance slowdown for common cases.